### ISSUE TITLE
wsl update failed: update failed: updating wsl: exit code: 4294967295

### ISSUE THREAD

**Issue by JulianHopkins** [microsoft/WSL#11554]
### Windows Version

22H2 19045.2130

### WSL Version

5.10.102.1

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1

### Distro Version

Ubuntu-22.04

### Other Software

Docker 4.29.0

### Repro Steps

After installing Docker Desktop for Windows version 4.29.0, during the first launch it tries to update WSL and an error occurs

### Expected Behavior

Launching Docker Desktop for Windows version 4.29.0 without error

### Actual Behavior

After installing Docker Desktop for Windows version 4.29.0, during the first launch it tries to update WSL and an error occurs:
wsl update failed: update failed: updating wsl: exit code: 4294967295: running WSL command wsl.exe C:\windows\System32\wsl.exe --update --web-download: (c) >@?>@0F8O 09:@>A>DB (Microsoft Corporation). A5 ?@020 70I8I5=K.

A?>;L7>20=85: wsl.exe [Argument] [Options...] [CommandLine]
...

### Diagnostic Logs

Diagnostics ID: DCA0E237-8CBF-4D04-A429-1780CAB53F79/20240504102559

**Comment by JulianHopkins** [comment]
[WslLogs-2024-05-04_14-39-40.zip](https://github.com/microsoft/WSL/files/15209568/WslLogs-2024-05-04_14-39-40.zip)

**Comment by JulianHopkins** [comment]
What does "Appx package is not installed" mean?

**Comment by JulianHopkins** [comment]
[WslLogs-2024-05-04_14-55-07.zip](https://github.com/microsoft/WSL/files/15209593/WslLogs-2024-05-04_14-55-07.zip)

**Comment by JulianHopkins** [comment]
[WslLogs-2024-05-04_15-16-08.zip](https://github.com/microsoft/WSL/files/15209674/WslLogs-2024-05-04_15-16-08.zip)

**Comment by Grouane** [comment]
Did you manage to solve the issue ?
I have the same issue and can't find anything to solve it

**Comment by JulianHopkins** [comment]
No, it didn't work. Even complete removal and disabling of the VSL
subsystem did not help. I can’t move beyond version 5.10.102.1. Although
new, as far as I know: 5.15.146.1-2

пн, 6 мая 2024 г. в 14:41, Grouane ***@***.***>:

**Comment by OneBlue** [comment]
Thank you @JulianHopkins. Does manually installing the [MSI package](https://github.com/microsoft/WSL/releases/download/2.2.4/wsl.2.2.4.0.x64.msi) solve the issue ?

**Comment by JulianHopkins** [comment]
No

**Comment by oeon** [comment]
Applying this Windows update allowed me to update to 4.30 https://github.com/docker/for-win/issues/14022#issuecomment-2065838496 maybe it will help you too ¯\_(ツ)_/¯

**Comment by vinalian** [comment]
> Thank you @JulianHopkins. Does manually installing the [MSI package](https://github.com/microsoft/WSL/releases/download/2.2.4/wsl.2.2.4.0.x64.msi) solve the issue ?

Thank you, it helped me.

**Comment by karema9** [comment]
wsl update failed: update failed: updating wsl: exit code: 4294967295: running WSL command wsl.exe C:\Windows\System32\wsl.exe --update --web-download: Copyright (c) Microsoft Corporation. All rights reserved.



Usage: wsl.exe [Argument] [Options...] [CommandLine]



Arguments for running Linux binaries:



    If no command line is provided, wsl.exe launches the default shell.



    --exec, -e <CommandLine>

        Execute the specified command without using the default Linux shell.



    --

        Pass the remaining command line as is.



Options:

    --cd <Directory>

        Sets the specified directory as the current working directory.

        If ~ is used the Linux user's home path will be used. If the path begins

        with a / character, it will be interpreted as an absolute Linux path.

        Otherwise, the value must be an absolute Windows path.



    --distribution, -d <Distro>

        Run the specified distribution.



    --user, -u <UserName>

        Run as the specified user.



    --system

        Launches a shell for the system distribution.



Arguments for managing Windows Subsystem for Linux:



    --help

        Display usage information.



    --install [Options]

        Install additional Windows Subsystem for Linux distributions.

        For a list of valid distributions, use 'wsl --list --online'.



        Options:

            --distribution, -d [Argument]

                Downloads and installs a distribution by name.



                Arguments:

                    A valid distribution name (not case sensitive).



                Examples:

                    wsl --install -d Ubuntu

                    wsl --install --distribution Debian



    --set-default-version <Version>

        Changes the default install version for new distributions.



    --shutdown

        Immediately terminates all running distributions and the WSL 2

        lightweight utility virtual machine.



    --status

        Show the status of Windows Subsystem for Linux.



    --update [Options]

        If no options are specified, the WSL 2 kernel will be updated

        to the latest version.



        Options:

            --rollback

                Revert to the previous version of the WSL 2 kernel.



Arguments for managing distributions in Windows Subsystem for Linux:



    --export <Distro> <FileName>

        Exports the distribution to a tar file.

        The filename can be - for standard output.



    --import <Distro> <InstallLocation> <FileName> [Options]

        Imports the specified tar file as a new distribution.

        The filename can be - for standard input.



        Options:

            --version <Version>

                Specifies the version to use for the new distribution.



    --list, -l [Options]

        Lists distributions.



        Options:

            --all

                List all distributions, including distributions that are

                currently being installed or uninstalled.



            --running

                List only distributions that are currently running.



            --quiet, -q

                Only show distribution names.



            --verbose, -v

                Show detailed information about all distributions.



            --online, -o

                Displays a list of available distributions for install with 'wsl --install'.



    --set-default, -s <Distro>

        Sets the distribution as the default.



    --set-version <Distro> <Version>

        Changes the version of the specified distribution.



    --terminate, -t <Distro>

        Terminates the specified distribution.



    --unregister <Distro>

        Unregisters the distribution and deletes the root filesystem.



    --mount <Disk>

        Attaches and mounts a physical disk in all WSL2 distributions.



        Options:

            --bare

                Attach the disk to WSL2, but don't mount it.



            --type <Type>

                Filesystem to use when mounting a disk, if not specified defaults to ext4.



            --options <Options>

                Additional mount options.



            --partition <Index>

                Index of the partition to mount, if not specified defaults to the whole disk.



    --unmount [Disk]

        Unmounts and detaches a disk from all WSL2 distributions.

        Unmounts and detaches all disks if called without argument.



I am getting the same erro while starting Docker Desktop on Windows. I have tried the recommended solutions with no success, what could be the issue.

**Comment by Traineratwot** [comment]
GPT: 
The error you're encountering is related to updating the Windows Subsystem for Linux (WSL). Error code 1603 typically indicates a general installation failure, while exit code 4294967295 (or 0xffffffff) may signify that an error occurred during the execution of the command.

Here are several steps that may help you resolve this issue:

1. **Check the Installation Log**: Open the log file mentioned in the error message (`<HOME>\AppData\Local\Temp\wsl-install-logs.txt`) to get more detailed information about what went wrong.

2. **Update Windows**: Ensure that you have all the latest Windows updates installed. Sometimes, WSL issues can be related to outdated Windows components.

3. **Restart Your Computer**: Sometimes a simple reboot can help resolve temporary issues.

4. **Reinstall WSL**:
   - Open PowerShell as an administrator and run the following command to unregister WSL:
     ```powershell
     wsl --unregister <DistroName>
     ```
     Replace `<DistroName>` with the name of your distribution (e.g., Ubuntu).
   - Then reinstall WSL using:
     ```powershell
     wsl --install
     ```

5. **Check for Conflicting Programs**: Some antivirus programs or system utilities may block the installation of updates. Try temporarily disabling them and attempt the update again.

6. **Run the Installation Manually**: If you know which version of WSL you need, you can try downloading and installing it manually from the official Microsoft website.

7. **Use the `wsl --update` Command**: Try executing the update command manually in PowerShell as an administrator:
   ```powershell
   wsl --update
   ```

If none of these steps work, you may want to seek help on the official Microsoft forums or the Windows community for more specific guidance.

---

It helped me
